public/app/features/dashboard/state/PanelQueryRunner.ts

```diff
@@ -174,10 +186,20 @@ export class PanelQueryRunner {
     this.lastResult = data;
   };
 
+  resendLastResult = () => {
+    if (this.lastResult) {
+      this.subject.next(this.lastResult);
+    }
+  };
+
   setTransformations(transformations?: DataTransformerConfig[]) {
     this.transformations = transformations;
   }
 
+  setFieldOverrides = (options?: ApplyFieldOverrideOptions) => {
+    this.fieldOverrideOptions = options;
+  };
```

Comment:
I have an idea of instead of these setters have PanelQueryRunner take in the constructor a DataConfigSource 

```typescript
interface DataConfigSource {
  getTransformations: () => DataTransformationsConfig;
  getFieldOverrideOptions: () => ApplyFieldOverrideOptions; 
}
```

This feels cleaner as the user (PanelModel) does not have to constantly update internal state in PanelQueryRunner when transformations change or overrides change. Just need to resend (reprocess) last result and it will ask for the current transformations & overrides.

Reply:
I like this idea. Implemented in 98bd819 and it simplifies a lot